### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by tanaes**
Added dox for adding modules @antgonza

### CODE REVIEW

File: doc/contributing.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Contributing a module to *Oecophylla*
```
Comment (antgonza) [inline doc/contributing.md:1]: Is this capitalization OK and respected in other places in the documentation?

File: doc/contributing.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Check the [Snakemake documentation](http://snakemake.readthedocs.io) for 
+specifics on formatting Snakemake rules.
+
+
+## The module environment
+
+Rules can be executed in their own environment, allowing tools with potentially
+conflicting dependences to execute in the workflow.
+
+In `oecophylla`, we are specifying the environment for a rule's execution with
```
Comment (antgonza) [inline doc/contributing.md]: capitalization used at the beginning not used here ...

**Review by antgonza — COMMENTED** [review]
a couple of comments!

File: doc/contributing.md (hunk 1 above)
Comment (tanaes) [reply doc/contributing.md:1]: Do we have a 'house style' for capitalization?

Comment (antgonza) [reply doc/contributing.md:1]: No? But I think this is the only place where the O is capitalized ...

Comment (antgonza) [reply doc/contributing.md:1]: BTW, could you also check the README so it follows this?

Comment (tanaes) [reply doc/contributing.md:1]: 👍

File: doc/contributing.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+In Oecophylla, we are specifying the environment for a rule's execution with
+a line at the beginning of a shell execution block that executes the commands
+necessary to initialize that environment. The necessary command is passed into
+the `params` object in the rule specification from the `config.yaml` file.
+This enables the use of multiple mechanisms for specifying environments -- for
+example, a centralized install on a cluster could use `module load foo` while
+a local test installation could use `source activate oecophylla-foo`. 
+
+To specify the environment, make sure to access the corresponding declration
```
Comment (wasade) [inline doc/contributing.md]: declration ->declaration

File: doc/contributing.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+```yaml
+
+rule foo1:
+    input:
+        file.txt
+    output:
+        file.foo.txt
+    params:
+        env = config['envs']['foo']
```
Comment (wasade) [inline doc/contributing.md:77]: Is the expectation that `config['envs']['foo']` is of type `str`, and that `str` describes a shell command (e.g., `source activate foo`)? Perhaps a comment line?

**Review by wasade — COMMENTED** [review]
Just some small comments

**Pull request merged**